src/reservoir/Reservoir.ts

```diff
@@ -211,6 +212,67 @@ export class Reservoir {
     };
   }
 
+  async generateFulfillOrderExecutionData({
+    askOrBid,
+    signature,
+    tokenId,
+  }: {
+    askOrBid: SeaportAskOrBid;
+    signature: Hash;
+    tokenId: bigint;
+  }) {
+    const order = {
+      parameters: {
+        ...askOrBid.rawData,
+        totalOriginalConsiderationItems: BigInt(
+          askOrBid.rawData.consideration.length
+        ),
+      },
+      numerator: 1n,
+      denominator: 1n,
+      signature,
+      extraData: "0x" as Hash,
+    };
+
+    const nftWithCriteriaIndex = order.parameters.consideration.findIndex(
+      (consid) => consid.itemType === 4
+    );
+
+    const fulfillOrderCallbackData = encodeFunctionData({
+      abi: seaportABI,
+      functionName: "fulfillAdvancedOrder",
+      args: [
+        order,
+        nftWithCriteriaIndex !== -1
```

Comment:
Should we fail in this case?

Reply:
There are some bids that do not have an NftWithCriteria, like private bids